cmd/releasego/update-azure-linux.go

```diff
@@ -61,118 +80,190 @@ func updateAzureLinux(p subcmd.ParseFunc) error {
 		return err
 	}
 
-	assets, err := loadBuildAssets(buildAssetJSON)
-	if err != nil {
-		return err
-	}
+	// Set custom user agent to help GitHub identify the bot if necessary.
+	client.UserAgent = "microsoft/go-infra update-azure-linux"
 
-	if updateBranch == "nil" || updateBranch == "" {
-		updateBranch = generateUpdateBranchNameFromAssets(assets)
-	}
-
-	golangSpecFileBytes, err := downloadFileFromRepo(ctx, client, owner, repo, baseBranch, golangSpecFilepath)
+	// Check that the PAT has the necessary scopes.
+	patScopes, err := githubutil.PATScopes(ctx, client)
 	if err != nil {
 		return err
 	}
+	if !slices.Contains(patScopes, "repo") {
+		return fmt.Errorf("the PAT must have 'repo' scope, but not found in list: %v", patScopes)
+	}
 
-	golangSpecFileContent := string(golangSpecFileBytes)
+	if upstream != owner {
+		// Submitting PR via fork. Try to make sure it'll work.
+		if !slices.Contains(patScopes, "workflow") {
+			return fmt.Errorf("the PAT must have 'workflow' scope, but not found in list: %v", patScopes)
+		}
 
-	prevGoArchiveName, err := extractGoArchiveNameFromSpecFile(golangSpecFileContent)
-	if err != nil {
-		return err
+		// Check if fork already exists, or create it.
```

Comment:
nit: This comment confused me. `FetchRepositoryOrNil` doesn't create the fork if it doesn't exist. It is done some lines below by `FullyCreateFork`. Consider adding a blank line below the comment so it is more clear that it applies to the subsequent lines, or split the comment in two: `Check if fork already exists` and `the fork doesn't exist, create it` just before `FullyCreateFork`.